paf-mvp-core-js/src/ui/fields.ts

```diff
@@ -99,24 +158,62 @@ export abstract class Field<T, M extends IModel> extends FieldReadOnly<T, M> imp
   }
 
   /**
-   * Sets the current value, updating any HTML elements that match the name value, then checks to see if any other
-   * fields need to be updated if the model is not already in a setting values operation.
+   * Sets the current value, then checks to see if any other fields need to be updated if the model is not already in a
+   * setting values operation. Finally updates any UI elements bound to the model.
+   * @remarks if the field is disabled an exception is thrown.
    */
   public set value(value: T) {
+    if (this._disabled) {
+      throw "Can't set value of disabled field";
+    }
     this._value = value;
-    this.bindings.forEach((b) => b.setValue(value));
     if (this.model.settingValues === false) {
       this.model.settingValues = true;
       this.updateOthers();
       this.model.settingValues = false;
     }
+    this.model.updateUI();
+  }
+
+  /**
+   * Gets the value that was persisted, or undefined if there is no persisted value.
+   */
+  public get persistedValue(): T {
```

Comment:
NIT: adding `|undefined` might help detect weaknesses in the rest of the project:
```typescript
  public get persistedValue(): T | undefined {
    return this._value;
  }
```